src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -71,7 +75,7 @@ ILoggerFactory loggerFactory
 
         public IEnumerable<ServiceEndpoint> GetOnlineEndpoints()
         {
-            return ConnectionContainers.Keys.Where(s => s.Online);
+            return _connectionContainers.Keys.Where(s => s.Online);
```

Comment:
>Keys [](start = 41, length = 4)

for concurrent dict, use `dict.Where(s=>s.Key...)`, `.Keys` lock the dict instead of the bucket